framework/include/base/MooseObject.h

```diff
@@ -82,6 +152,9 @@ class MooseObject :
   /// The name of this object, reference to value stored in InputParameters
   const std::string & _name;
 
+  /// Reference to the "enable" InputParaemters, used by Controls for toggling on/off MooseObjects
+  const bool & _enabled;
```

Comment:
Perhaps, but do you trust the user to make this decision? If you disable a UO, Postprocessor, Kernel, BC, Material even an AuxKernel. You risk having the user make the mistake of not handling the "stale" values that would be received through the coupling. I can see a hundred ways for this to go bad but only very few ways for it to work properly. This might be a support nightmare.